spec/lib/newspaper_works/ingest/ndnp/page_ingester_spec.rb

```diff
@@ -91,17 +91,68 @@
   describe "file import integration" do
     do_now_jobs = [IngestLocalFileJob, IngestJob, InheritPermissionsJob]
 
-    it "attaches primary, derivative files", perform_enqueued: do_now_jobs do
-      adapter.ingest
-      page = adapter.target
+    let(:issue_data) do
+      NewspaperWorks::Ingest::NDNP::IssueIngest.new(issue2)
+    end
+
+    let(:page_data_minus_tiff) { issue_data.to_a[0] }
+
+    def check_fileset(page)
       fileset = page.members.select { |m| m.class == FileSet }[0]
       # Reload fileset because jobs have modified:
       fileset.reload
       expect(fileset).not_to be_nil
       expect(fileset.original_file).not_to be_nil
       expect(fileset.original_file.mime_type).to eq 'image/tiff'
-      derivatives = NewspaperWorks::Data::WorkDerivatives.new(page, fileset)
+      expect(fileset.original_file.size).to be > 0
+    end
+
+    def expect_file_assignment_logging(adapter)
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |v| v.include?('Assigned primary file to work') }
+      ).once
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |v| v.include?('Assigned derivative file to work') }
+      ).exactly(3).times
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |v| v.include?('Beginning file attachment') }
+      ).once
+    end
+
+    def expect_page_import_logging(adapter)
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |v| v.include?('Created NewspaperPage work') }
+      ).once
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |v| v.include?('Saved metadata to NewspaperPage work') }
+      ).once
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |v| v.include?('Linked NewspaperIssue') }
+      ).once
+    end
+
+    it "attaches primary, derivative files", perform_enqueued: do_now_jobs do
+      expect_page_import_logging(adapter)
+      expect_file_assignment_logging(adapter)
+      adapter.ingest
+      page = adapter.target
+      check_fileset(page)
+      derivatives = NewspaperWorks::Data::WorkDerivatives.new(page)
       expect(derivatives.keys).to match_array ["jp2", "xml", "pdf"]
     end
+
+    # support this use-case for evaluation purposes
+    it "generates TIFF when missing from page", perform_enqueued: do_now_jobs do
+      adapter = described_class.new(page_data_minus_tiff, issue)
+      expect_page_import_logging(adapter)
+      expect(adapter).to receive(:write_log).with(
+        satisfy { |arg| arg.include?('Creating TIFF') },
+        Logger::WARN
+      ).exactly(1).times
+      expect_file_assignment_logging(adapter)
+      adapter.ingest
+      # expect { adapter.ingest }.not_to raise_error
```

Comment:
remove commented line (unless needed for future reference?)

Reply:
Good catch — this was accidental, that test (however superfluous it could be to fail on error) passes (originally commented out because it looked like logging tests were not playing well with running ingest method in a block, but they do).  I have re-enabled this line and removed the line before it.  Fixed in f40bf67762537a0abd84dbeadbb11144cf9972d6